packages/@webex/plugin-meetings/src/meeting/util.ts

```diff
@@ -180,7 +180,7 @@ const MeetingUtil = {
       .then((res) => {
         const parsed = MeetingUtil.parseLocusJoin(res);
         meeting.setLocus(parsed);
-
+        meeting.isoLocalClientMeetingJoinTime = res?.headers?.date; // read from header if exist, else fall back to system clock : https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-555657
```

Comment:
wouldn't it be better to add this to the common code that's always executed when any Locus is parsed? I think we are not guaranteed to have the join time in the http response here, it might come in a delta DTO afterwards. Actually if we always parse it out of Locus DTO then we would end up setting it also in the case when we just load the page and Locus thinks we're already in the meeting, because user was in a meeting and just did a page refresh, so maybe it is better to keep it here as it is and my comment is useless - just making this comment to make sure that all these scenarios were considered.

Also just wondering if there are any other means of ending up in the meeting where we should set the joined time, like maybe for example Meeting.moveFrom()?

Reply:
I think this is maybe a misunderstanding in how we are supposed to be "setting meeting joined time" according to the spec, we are *only* supposed to set `meetingJoinedTime` when we receive a locus join response (from `/participant`, or `/call` request). The time is not set in the DTO or anything else, and it's not the system clock. the request is to use the value exposed in the `Date` header field from the join response. We would *not* want the one that comes in a delta DTO afterwards. In order to support this, Locus has added `Date` to `Access-Control-Expose-Headers`.

I *believe* all the "join" functions route through this location. At least every one that I checked did (for example `moveFrom` will also end up here)